library/clinical_rules.php

```diff
@@ -1042,7 +1045,37 @@ function test_rules_clinic($provider = '', $type = '', $dateTarget = '', $mode =
                     }
 
                     // Check if pass filter
-                    $passFilter = test_filter($rowPatient['pid'], $rowRule['id'], $dateFocus);
+                    /*
+                    HR: changed behavior of test_filter(), and will pass it $dateTarget instead of $dateFocus. Will also call test_filter() only once, not for each $dateFocus value
+                    test_filter() looks for patient data entered prior to $dateTarget timepoint
+
+                    test_filter() was returning
+                    false if an inclusion filter does not succeed
+                    "EXCLUDED" if an exclusion filter succeeds
+                    otherwise true
+
+                    Changed so it now returns:
+                    -- if any required inclusions fail, return false
+                    -- if there are no required inclusions, and some optional inclusions exist, and any optional inclusions succeed,
+                    and either exclusions don't exist or exclusions don't succeed, return true
+                    -- if all inclusions are optional, and none succeed, return false
+                    -- if there are no inclusions, and there are exclusions, and exclusions do not succeed, return true
+                    -- if exclusions succeed (checked only if there are no inclusions, or if inclusions succeed), return 'EXCLUDED'
+                    -- if no inclusions or exclusions, return false
+
+                    -- when processing inclusions, if filters exist in multiple categories (e.g. age, gender and lifestyle), need to process all categories.
+                    -- If required filters in one category succeed, need to check for required filters in other categories
+                    -- Similarly, if all filters in one category are optional and do not succeed, need to see if optional filters exist in different category
+                    -- that might succeed
+
+                    -- Mixing optional and required filters makes no sense, but is tollerated. If one filter is required, any optional filters have no relevence
+
+                    -- Same ideas apply to analysis of targets as well as filters
+                    */
+                    if ($dateCounter == 1) { // only do once, for $dateTarget (which is generally "today")
+                        $passFilter = test_filter($rowPatient['pid'], $rowRule['id'], $dateTarget);
+                    }
```

Comment:
rec trying to move lines 1075 - 1094 (drop the last else clause) to below line 1028

Reply:
This then removes the filter logic outside the target datefocus stuff since no need to run that if that filter fails